### PR TITLE
feat: support ECDSA JWT signing.

### PR DESCRIPTION
**Pull request by farmborough**
fixes gravitee-io/issues#4059

### CODE REVIEW

File: gravitee-am-certificate/gravitee-am-certificate-api/src/main/java/io/gravitee/am/certificate/api/AbstractCertificateProvider.java [hunk 1]
```diff
@@ -87,12 +96,16 @@ public void createCertificateKeys(CertificateMetadata certificateMetadata) throw
                     certificateKeys.add(new CertificateKey(CertificateFormat.PEM, pem));
                     expirationDate = ((X509Certificate) cert).getNotAfter();
                 }
-                certificateKeys.add(new CertificateKey(CertificateFormat.SSH_RSA, RSAKeyUtils.toSSHRSAString((RSAPublicKey) keyPair.getPublic())));
+                PublicKey publicKey = keyPair.getPublic();
+                if (publicKey.getAlgorithm().equals(RSA)){
+                    certificateKeys.add(new CertificateKey(CertificateFormat.SSH_RSA, RSAKeyUtils.toSSHRSAString((RSAPublicKey) publicKey)));
+                } else if (publicKey.getAlgorithm().equals(EC)){
+                    certificateKeys.add(new CertificateKey("ECSDA", toEcdsaString((ECPublicKey) publicKey)));
```
Comment (leleueri) [inline gravitee-am-certificate/gravitee-am-certificate-api/src/main/java/io/gravitee/am/certificate/api/AbstractCertificateProvider.java:103]: Can we create an entry into the CertificateFormat class ?

File: gravitee-am-certificate/gravitee-am-certificate-api/src/main/java/io/gravitee/am/certificate/api/AbstractCertificateProvider.java [hunk 2]
```diff
@@ -213,41 +236,66 @@ private JWK createRSAKey(com.nimbusds.jose.jwk.JWK nimbusJwk, boolean includePri
...
+        if (nimbusEcJwk.getCurve() != null) {
+            jwk.setCrv(nimbusEcJwk.getCurve().toString());
+        }
+        if (nimbusEcJwk.getX() != null) {
+            jwk.setX(nimbusEcJwk.getX().toString());
+        }
+        if (nimbusEcJwk.getY() != null) {
+            jwk.setY(nimbusEcJwk.getY().toString());
+        }
+        if (nimbusEcJwk.getCurve() != null) {
```
Comment (leleueri) [inline gravitee-am-certificate/gravitee-am-certificate-api/src/main/java/io/gravitee/am/certificate/api/AbstractCertificateProvider.java:258]: duplicate with line 249 ?

**Review by leleueri — CHANGES_REQUESTED** [review]

File: gravitee-am-certificate/gravitee-am-certificate-api/src/main/java/io/gravitee/am/certificate/api/AbstractCertificateProvider.java (hunk 2 above)
Comment (farmborough) [reply gravitee-am-certificate/gravitee-am-certificate-api/src/main/java/io/gravitee/am/certificate/api/AbstractCertificateProvider.java:258]: Oops.

**Review by leleueri — CHANGES_REQUESTED** [review]
I'm not able to use EC algorithm to generate default domain certificate (used when a domain is created)
https://github.com/gravitee-io/gravitee-access-management/blob/master/gravitee-am-management-api/gravitee-am-management-api-standalone/gravitee-am-management-api-standalone-distribution/src/main/resources/config/gravitee.yml#L205

Tried with :
* ECDSA
* SHA256withECDSA

Maybe my values were wrong but as we generate RSA keys, I think we will have trouble anyway :thinking: 
https://github.com/gravitee-io/gravitee-access-management/blob/master/gravitee-am-service/src/main/java/io/gravitee/am/service/impl/CertificateServiceImpl.java#L375
what is your thought

**Comment by farmborough** [comment]
> I'm not able to use EC algorithm to generate default domain certificate (used when a domain is created) https://github.com/gravitee-io/gravitee-access-management/blob/master/gravitee-am-management-api/gravitee-am-management-api-standalone/gravitee-am-management-api-standalone-distribution/src/main/resources/config/gravitee.yml#L205
> 
> Tried with :
> 
> * ECDSA
> * SHA256withECDSA
> 
> Maybe my values were wrong but as we generate RSA keys, I think we will have trouble anyway 🤔 https://github.com/gravitee-io/gravitee-access-management/blob/master/gravitee-am-service/src/main/java/io/gravitee/am/service/impl/CertificateServiceImpl.java#L375 what is your thought

Ah yes I think I can see the line in `CertificateServiceImpl` that needs changing (it's hardcoded to "RSA").

**Review by leleueri — APPROVED** [review]

**Pull request merged**